src/server/helpers/utils.js

```diff
@@ -49,11 +50,80 @@ export function getEntityModels(orm: Object): Object {
 		Work
 	};
 }
+function getRevisionModels(orm) {
+	const {AuthorRevision, EditionRevision, EditionGroupRevision, PublisherRevision, WorkRevision} = orm;
+	return [
+		AuthorRevision,
+		EditionGroupRevision,
+		EditionRevision,
+		PublisherRevision,
+		WorkRevision
+	];
+}
+async function 	getCompleteRevision(revision, orm) {
+	const {Entity} = orm;
+	const RevisionModels = getRevisionModels(orm);
+	for (let i = 0; i < RevisionModels.length; i++) {
+		const Model = RevisionModels[i];
+		// eslint-disable-next-line no-await-in-loop
+		const model = await new Model({id: revision.id})
+			.fetch({
+				withRelated: [
+					'data.aliasSet.defaultAlias'
+				]
+			});
+		if (model) {
+			const modelJSON = model.toJSON();
+			// eslint-disable-next-line no-await-in-loop
+			const entity = await new Entity({bbid: modelJSON.bbid}).fetch();
+			const entityJSON = entity ? entity.toJSON() : {};
```

Comment:
I used 78, 79 instead of  `const modelType = model.tableName.match(/^bookbrainz.(\w+)_revision$/i)[1];
			                const databaseModelName = _.upperFirst(_.camelCase(modelType));`

This seems less hacky to me.